ksql-serde/src/main/java/io/confluent/ksql/serde/util/SerdeProcessingLogMessageFactory.java

```diff
@@ -34,6 +36,20 @@ private SerdeProcessingLogMessageFactory() {
   public static Function<ProcessingLogConfig, SchemaAndValue> deserializationErrorMsg(
       final Throwable exception,
       final Optional<byte[]> record
+  ) {
+    return message(exception, () -> record.map(Base64.getEncoder()::encodeToString).orElse(null));
+  }
+
+  public static Function<ProcessingLogConfig, SchemaAndValue> deserializationErrorMsg(
```

Comment:
I'd use a different message type for this error. Deserialization error messages generally mean the serde couldn't deserialize the record in the topic, and contain a base64 encoded representation of the data inside (in a field called `recordB64`). The data written out here represents something different - KSQL's view of the row once it's been deserialized.

Reply:
You should be able to use the existing record processing error (`EngineProcessingLogMessageFactory.recordProcessingError`)

Reply:
i was debating which I should do since this felt like a type of deserialization error to me, but I'll update it to a new type (I can see both sides)